app/services/application_service.rb

```diff
@@ -0,0 +1,7 @@
+class ApplicationService
+  include Dry::Monads[:result]
+
+  def self.call(*args, &block)
```

Comment:
I suggest to use [argument forwarding](https://blog.saeloun.com/2020/09/16/ruby-3-0-to-add-support-for-forwarding-arguments-along-with-lead-arguments/)